### PR TITLE
Implemented heterogeneity correction using APM base data

### PR DESCRIPTION
**Pull request by AdmiralHomes**
Based on Johannas work on proton heterogeneity correction, it was also implemented for carbon ions, using Hans-Peters fitted APM alpha beta curves for RBE. 

Switching on the correction is done via fields in the pln:

1. pln.heterogeneity.calcHetero = true;  _% generally enables correction_
2. pln.heterogeneity.type = 'complete';  _% optional ('complete','voxelwise','depthbased')_
3. pln.heterogeneity.useDoseCurves = true;  _% enables the usage of the fitted alpha and beta curves_
4. pln.heterogeneity.useOrgDepths = false;  _% Switch for using the original depth doses (no heterogeneity correction possible)_

For better testing, a modified boxphantom is introduced, which features a 'lung' slab with height=60mm in front of the body. The resulting RBE dose was plotted in the central axial slice (z=240mm). The calculation used the above mentionend parameters. The dose difference plot shows the expected differences at the distal edge of the tumor.

withoutCorrection
withCorrection
doseDifference

### CODE REVIEW

File: heterogeneityEval.m [hunk 1]
```diff
@@ -0,0 +1,145 @@
+function [dose] = heterogeneityEval(particleType, doseMode, exportFolder, patients)
+% Heterogeneity correction evaluation script
+%
+% Computes the dose for the HIT plan (recalc), for protons and for carbons
+% with and without heterogeneity correction. Saves the workspace for each
```
Comment (becker89) [inline heterogeneityEval.m]: Maybe you can move these files into the tools subfolder ? I think that's a better place for the files.

File: matRad_calcParticleDose.m [hunk 2]
```diff
@@ -34,40 +34,54 @@
...
+
+% load machine file
+fileName = [pln.radiationMode '_' pln.machine];
+try
+    load([fileparts(mfilename('fullpath')) filesep fileName]);
+catch
+    matRad_dispToConsole(['Could not find the following machine file: ' fileName ],param,'error');
+end
+
+pln = matRad_checkHeterogeneity(pln,cst,machine.data(1));
```
Comment (becker89) [inline matRad_calcParticleDose.m]: Does this function also work if the heterogeneity correction is turned off ?

File: matRad_calcParticleDose.m [hunk 3]
```diff
@@ -186,471 +200,530 @@
...
-   % only if LEM is used corresponding bio data must be available in the base data set
-   if strcmp(pln.bioParam.model,'LEM')
-      if isfield(machine.data,'alphaX') && isfield(machine.data,'betaX')
-         
-         fprintf('matRad: loading biological base data... ');
-         
-         for i = 1:size(cst,1)
-            
+    
+    dij.alphaX  = zeros(dij.doseGrid.numOfVoxels,1);
```
Comment (becker89) [inline matRad_calcParticleDose.m]: You like alphaX and betaX more that ax and bx?

File: tools/matRad_cstHeteroAutoassign.m [hunk 4]
```diff
@@ -0,0 +1,13 @@
+function [cstHetero] = matRad_cstHeteroAutoassign(cst)
+
```
Comment (becker89) [inline tools/matRad_cstHeteroAutoassign.m]: header missing

File: tools/matRad_plotSOBP.m [hunk 5]
```diff
@@ -0,0 +1,38 @@
+function matRad_plotSOBP(ct, stf, cube1, cube2)
+if exist('cube2')
```
Comment (becker89) [inline tools/matRad_plotSOBP.m]: header missing

File: matRad_calcParticleDose.m (hunk 2 above)
Comment (AdmiralHomes) [reply matRad_calcParticleDose.m]: yes, the matRad_checkHeterogeneity function should select the required parameters only if the correction enabled, in all other cases, calcParticleDose works without restrictions.

File: matRad_calcParticleDose.m (hunk 3 above)
Comment (AdmiralHomes) [reply matRad_calcParticleDose.m]: You started this in your APM branch and I thought it's clearer to read.

File: tools/matRad_compareDose.m [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+
+if (~exist('enable','var') || isempty(enable)) || strcmp(enable,'all')
+    enable = [1 1 1];
+end
+if enable(1)==0
+    gammaCube=[];
+    gammaPassRate=[];
+end
+%% Load colormap for difference
+load('diffCMap.mat');
```
Comment (wahln) [inline tools/matRad_compareDose.m]: This should follow the standard matRad colormap worklfow. Add an diffCMap.m file to the colormaps folder, such that it get's recognized when using the matRad_getColormap function, and use it here.

**Pull request closed without merging**